apps/homescreen/js/app.js

```diff
@@ -809,6 +811,18 @@ const SETTINGS_VERSION = 0;
 
       case 'hashchange':
         if (!document.hidden) {
+          // If a dialog is showing, cancel the dialog
+          for (var dialog of this.dialogs) {
+            if (dialog.style.display === 'none') {
```

Comment:
Wouldn't be better to use the gaia-dialog component methods (.opened) instead of checking the style?